test/end-to-end/thirdpartyemailpassword.test.js

```diff
@@ -440,4 +442,498 @@ describe("SuperTokens Third Party Email Password", function () {
             assert.strictEqual(getAuthFromBackendInStorage, "true");
         });
     });
+
+    describe("SignIn default field tests", function () {
+        it("Should contain email and password fields prefilled", async function () {
+            await page.evaluate(() => window.localStorage.setItem("SHOW_SIGNIN_DEFAULT_FIELDS", "YES"));
+
+            await page.reload({
+                waitUntil: "domcontentloaded",
+            });
+
+            const expectedDefaultValues = {
+                email: "abc@xyz.com",
+                password: "fakepassword123",
+            };
+
+            const emailInput = await getInputField(page, "email");
+            const defaultEmail = await emailInput.evaluate((f) => f.value);
+            assert.strictEqual(defaultEmail, expectedDefaultValues["email"]);
+
+            const passwordInput = await getInputField(page, "password");
+            const defaultPassword = await passwordInput.evaluate((f) => f.value);
+            assert.strictEqual(defaultPassword, expectedDefaultValues["password"]);
+        });
+
+        it("Check on blank form submit nonOptionalErrorMsg gets displayed as expected", async function () {
+            await page.evaluate(() => window.localStorage.removeItem("SHOW_SIGNIN_DEFAULT_FIELDS"));
+
+            // set cookie and reload which loads the form with custom field
+            await page.evaluate(() =>
+                window.localStorage.setItem("SHOW_SIGNIN_WITH_NON_OPTIONAL_ERROR_MESSAGE", "YES")
+            );
+            await page.reload({
+                waitUntil: "domcontentloaded",
+            });
+
+            await waitForSTElement(page);
+            let apiCallMade = false;
+
+            await page.setRequestInterception(true);
+
+            const requestHandler = (request) => {
+                const url = request.url();
+                if (url === SIGN_IN_API) {
+                    apiCallMade = true;
+                    request.continue();
+                } else {
+                    request.continue();
+                }
+            };
```

Comment:
Please check but, I think `on("request", ...` fires without `setRequestInterception(true)`. We only need it to change/delay the responses.